SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialStatics.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Kismet/BlueprintFunctionLibrary.h"
+#include "SpatialStatics.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialStatics, Log, All);
+
+UCLASS()
+class SPATIALGDK_API USpatialStatics : public UBlueprintFunctionLibrary
+{
+	GENERATED_BODY()
+
+public:
+	/**
+	 * Returns whether SpatialOS networking has been enabled or not.
+	 */
+	UFUNCTION(BlueprintPure, Category = "SpatialOS")
+	static bool SpatialNetworkingEnabled();
+
+	/**
+	 * Returns true iff SpatialOS networking AND offloading are both enabled.
+	 */
+	UFUNCTION(BlueprintPure, Category = "SpatialOS")
+	static bool SpatialOffloadingEnabled();
+
+	/**
+	 * If SpatialOS networking is disabled, or if SpatialOS networking is enabled but offloading is disabled,
+	 * this will return Actor::HasAuthority(). If SpatialOS networking AND offloading is enabled, this will
+	 * returns true iff actor group worker association matches the current worker type.
+	 */
+	UFUNCTION(BlueprintPure, Category = "SpatialOS")
+	static bool IsAuthoritativeWorkerType(const AActor* Actor);
```

Comment:
This currently has inconsistent behaviour; authoritative worker and worker type aren't quite the same thing. If offloading is disabled then this should presumably always return true.

Reply:
A reason you'd want a function like this is - can this worker spawn this actor type, or can it send this type of RPC. That shouldn't rely on you currently having authority (even though in offloading that is what it means)

Reply:
Shall I change to `IsAuthoritativeWorker` with the knowledge that it will not work for offloaded and zoned workers

If not, what do you suggest?

Reply:
For a given actor we can already have a function that says if we have authority, so I'd suggest if we are adding functions that we make it an - is this the type of worker that could conceivably have authority over this actor.

Reply:
Do we have a use case for this being added?

Reply:
Yes we have a use case in Scavengers, basically anytime you have an actor you want offloaded, you'd need to replace all instances of `HasAuthority` with this function. 

The benefit of this approach vs. a "could conceivably have authority over this actor" is that user code doesn't need to care about an `is offloading enabled` check since it's encapsulated in the functionality. We may find that the need arises for both, but currently I think the need only exists for this function.

Reply:
Why doesn't HasAuthority work?

Reply:
I may have misunderstood something as I don't see how it can encapsulate both functions. They return different values in different cases.

Reply:
My understanding right now is that this will do the wrong thing if you created the actor and are waiting for the entity to be created or in zoning.